src/components/MoneyRequestReportView/MoneyRequestReportActionsList.tsx

```diff
@@ -206,72 +206,115 @@ function MoneyRequestReportActionsList({report, reportActions = [], transactions
+     * the MSG_VISIBLE_THRESHOLD), the unread marker will display over those new messages rather than the initial
+     * lastReadTime.
+     */
+    useLayoutEffect(() => {
+        if (unreadMarkerReportActionID) {
+            return;
+        }
 
-        // Scan through each visible report action until we find the appropriate action to show the unread marker
-        for (let index = startIndex; index < visibleReportActions.length; index++) {
-            const reportAction = visibleReportActions.at(index);
+        const mostRecentReportActionCreated = lastAction?.created ?? '';
+        if (mostRecentReportActionCreated <= unreadMarkerTime) {
+            return;
+        }
 
-            // eslint-disable-next-line react-compiler/react-compiler
-            if (reportAction && shouldDisplayNewMarker(reportAction, index)) {
-                return reportAction.reportActionID;
+        setUnreadMarkerTime(mostRecentReportActionCreated);
+
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, [lastAction?.created]);
+
+    const scrollToBottomForCurrentUserAction = useCallback(
+        (isFromCurrentUser: boolean) => {
+            InteractionManager.runAfterInteractions(() => {
+                setIsFloatingMessageCounterVisible(false);
+                // If a new comment is added from the current user, scroll to the bottom, otherwise leave the user position unchanged
+                if (!isFromCurrentUser) {
+                    return;
+                }
+
+                // We want to scroll to the end of the list where the newest message is
+                // however scrollToEnd will not work correctly with items of variable sizes without `getItemLayout` - so we need to delay the scroll until every item rendered
+                setTimeout(() => {
+                    reportScrollManager.scrollToEnd();
+                }, DELAY_FOR_SCROLLING_TO_END);
+            });
+        },
+        [reportScrollManager],
+    );
+
+    useEffect(() => {
+        // This callback is triggered when a new action arrives via Pusher and the event is emitted from Report.js. This allows us to maintain
+        // a single source of truth for the "new action" event instead of trying to derive that a new action has appeared from looking at props.
+        const unsubscribe = subscribeToNewActionEvent(report.reportID, scrollToBottomForCurrentUserAction);
+
+        return () => {
+            if (!unsubscribe) {
+                return;
             }
-        }
+            unsubscribe();
+        };
 
-        return null;
-    }, [currentUserAccountID, earliestReceivedOfflineMessageIndex, prevVisibleActionsMap, visibleReportActions, unreadMarkerTime]);
-    prevUnreadMarkerReportActionID.current = unreadMarkerReportActionID;
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
```

Comment:
Same here `react-hooks/exhaustive-deps` should be used only if there is no other viable option so leave comment explaining why it has to be skipped if there is no other way to handle this, thanks! (I know this code might have been reused but still)

Reply:
addressed